homeassistant/components/zha/logbook.py

```diff
@@ -0,0 +1,76 @@
+"""Describe ZHA logbook events."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from typing import TYPE_CHECKING
+
+import voluptuous as vol
+
+from homeassistant.components.logbook.const import (
+    LOGBOOK_ENTRY_MESSAGE,
+    LOGBOOK_ENTRY_NAME,
+)
+from homeassistant.const import ATTR_DEVICE_ID
+from homeassistant.core import Event, HomeAssistant, callback
+import homeassistant.helpers.device_registry as dr
+
+from .core.const import DOMAIN as ZHA_DOMAIN, ZHA_EVENT
+from .core.helpers import async_get_zha_device
+
+if TYPE_CHECKING:
+    from .core.device import ZHADevice
+
+
+@callback
+def async_describe_events(
+    hass: HomeAssistant,
+    async_describe_event: Callable[[str, str, Callable[[Event], dict[str, str]]], None],
+) -> None:
+    """Describe logbook events."""
+    device_registry = dr.async_get(hass)
+
+    @callback
+    def async_describe_zha_event(event: Event) -> dict[str, str]:
+        """Describe zha logbook event."""
+        device: dr.DeviceEntry = device_registry.devices[event.data[ATTR_DEVICE_ID]]
+        zha_device: ZHADevice = async_get_zha_device(hass, event.data[ATTR_DEVICE_ID])
```

Comment:
If the device has been deleted, will this raise ?

Reply:
good catch!